crates/turborepo-lib/src/gitignore.rs

```diff
@@ -0,0 +1,141 @@
+#[cfg(unix)]
+use std::os::unix::fs::PermissionsExt;
+use std::{
+    fs::{File, OpenOptions},
+    io,
+    io::{BufRead, Write},
+};
+
+use turbopath::AbsoluteSystemPathBuf;
+
+const TURBO_GITIGNORE_ENTRY: &str = ".turbo";
+const GITIGNORE_FILE: &str = ".gitignore";
+
+pub fn ensure_turbo_is_gitignored(repo_root: &AbsoluteSystemPathBuf) -> Result<(), io::Error> {
+    let gitignore_path = repo_root.join_component(GITIGNORE_FILE);
+
+    if !gitignore_path.exists() {
+        let mut gitignore = File::create(gitignore_path)?;
```

Comment:
We should prefer the operations on our path library over those from the stdlib. Primarily because we use [fs_err](https://docs.rs/fs-err/latest/fs_err/) which does us the huge favor of including paths in the error message.
```suggestion
        let mut gitignore = gitignore_path.create()?;
```